### PR TITLE
make beavis-ci work with multiple subdirectories and arbitrary repos

### PR DESCRIPTION
**Pull request by yymao**
@drphilmarshall here's my attempt to make your beavis-ci script work with multiple subdirectories (and also any repo). 

I have not yet tested this script to verify it works. That's on my to-do's. 

Looking at this script again, I wonder if we should in fact move this whole beavis-ci thing into its own repo? It really has no dependence on the DC2-analysis repo.

### CODE REVIEW

**Comment by yymao** [comment]
I've verified this script runs ok and works with multiple subdirectories. It also works on user-specified repo with the new option `-r`. As such, I believe it can be a standalone `beavis-ci ` repo as I mentioned above, and @drphilmarshall can set up one cron job for each repo that needs notebook rendering service.

File: beavis-ci/beavis-ci.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #   -u --username GITHUB_USERNAME, defaults to the environment variable
 #   -k --key      GITHUB_API_KEY, defaults to the environment variable
 #   -b --branch   Test the notebooks in a dev branch. Outputs still go to "rendered"
+#   -r --repo     Specify the repo name, default to LSSTDESC/DC2-analysis
```
Comment (drphilmarshall) [inline beavis-ci/beavis-ci.sh]: Nice! When we split beavis out, we'll need the default repo to be read from the $cwd, with, eg `repo=$( git config --get remote.origin.url | cut -d':' -f2 )`

File: beavis-ci/beavis-ci.sh [hunk 2]
```diff
@@ -107,65 +117,43 @@ else
...
-# Which notebooks to run? Either all of them, or the ones listed in README.rst:
-if [ $all -eq 0 ]; then
-    notebooks="$( grep '`ipynb' README.rst |& grep -v "No such file" | cut -d"<" -f2 | cut -d">" -f1 | sed s/'%20'/'_'/g | sed s/'%3A'/':'/g )" 
-fi
-# If README.rst is not present or empty of ipynb instances, default to running everything:
-if [ $all -gt 0 ] || [ ${#notebooks[0]} -eq 0 ]; then
-    notebooks="$( ls *.ipynb )"
-fi
 
+notebooks=`find . -path '*/.ipynb_checkpoints/*' -prune -o -name '*.ipynb' -print`
```
Comment (drphilmarshall) [inline beavis-ci/beavis-ci.sh]: I guess this is where we abandon parsing the README - I think that's OK given our plan to switch to writing READMEs automatically.

File: beavis-ci/beavis-ci.sh [hunk 3]
```diff
@@ -37,6 +38,9 @@ html=0
 all=0
 src="$0"
 branch='master'
+repo='LSSTDESC/DC2-analysis'
+jupyter='/usr/common/software/python/3.6-anaconda-4.4/bin/jupyter'
```
Comment (drphilmarshall) [inline beavis-ci/beavis-ci.sh]: I think we'll need a `-j` option to allow the user to select their own `jupyter` path, or use `which jupyter` to make `beavis-ci` system agnostic.

**Comment by drphilmarshall** [comment]
Thanks Yao! I'm going to merge this now, and then start a separate `beavis-ci` repo for us to do the README-generation in.

**Pull request merged**